specification/src/chapter3.adoc

```diff
@@ -1,45 +1,313 @@
+
+|===
+
+Interrupts: https://github.com/riscv/riscv-smmtt +
+External debug: https://github.com/riscv-non-isa/riscv-external-debug-security +
+Performance counters: 
+
+These extensions enable management of interrupts, external debug, and performance counters across supervisor domain boundaries:
+
+* External debug can be enabled for one supervisor domain without affecting other supervisor domains
+* M-mode firmware manage interrupt routing and preemption across supervisor domain boundaries
+* M-mode firmware can ensure that performance counters cannot be used by software in one supervisor domain to measure operations in other supervisor domains 
+
+==== MTT
+
+https://github.com/riscv/riscv-smmtt
+
+The _memory tracking table (MTT)_ is a memory structure managed by machine mode, tracking memory ownership across supervisor domains. It is designed to enable fine grained dynamic memory management across supervisor domain boundaries, with policy typically set by a hypervisor in a hosting domain responsible for resource management.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| PMP/ePMP MUST be used to protect M-mode from lower privilege levels
+
+| CAT_NNN  
+| MTT configurations MUST only be directly accessible to machine mode
+|===
+
+==== IOPMP
+
+https://github.com/riscv-non-isa/iopmp-spec
+
+IOPMP is a system level component providing physical memory access control for device initiated transactions, complementing PMP and sPMP rules. 
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| A system which supports PMP/ePMP, or sPMP, MUST implement IOPMP for device access control.
+
+| CAT_NNN
+| IOPMP configurations MUST only be directly accessible to machine mode.
+|===
+
+
+==== IOMTT
+
+https://github.com/riscv/riscv-smmtt
+
+IOMTT is a system level component providing physical memory access control for device initiated transactions, complementing MTT rules.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN  
+| A system which supports MTT MUST implement IOMTT for device initiated access control.
+
+| CAT_NNN
+| IOMTT configurations MUST only be directly accessible to machine mode.
+
+| CAT_NNN
+| A system which implements IOMTT MUST also implement IOPMP to protect device initiated accesses to M-mode memory.
+
+|===
+
+==== IOMMU
+
+https://github.com/riscv-non-isa/riscv-iommu
+
+IOMMU is a system level component providing virtual memory access for device initiated transactions, complementing MMU translation rules.
+
+[width=100%]
+[%header, cols="5,20"]
+|===
+| ID#     
+| Requirement
+
+| CAT_NNN
+| Systems supporting MMU SHOULD also support IOMMU
+
+| CAT_NNN  
+| Systems supporting IOMMU MUST also support IOPMP, an MAY support IOMTT, to enforce physical memory access control for device initiated transactions.
```

Comment:
I would argue that a system that support IOMMU *and* SMMTT should not require IOPMP.

Reply:
IOPMP or IOMTT suffices. Both are not required.